src/Azure.Functions.Cli/Common/AuthManager.cs

```diff
@@ -0,0 +1,312 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Functions.Cli.Actions.HostActions;
+using Azure.Functions.Cli.Arm.Models;
+using Azure.Functions.Cli.Helpers;
+using Azure.Functions.Cli.Interfaces;
+using Colors.Net;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using static Azure.Functions.Cli.Common.OutputTheme;
+using static Colors.Net.StringStaticMethods;
+
+namespace Azure.Functions.Cli.Common
+{
+    internal class AuthManager : IAuthManager
+    {
+        public AuthManager() { }
+
+        private AuthSettingsFile MiddlewareAuthSettings;
+
+        public async Task CreateAADApplication(string accessToken, string AADName, string appName)
+        {
+            if (string.IsNullOrEmpty(AADName))
+            {
+                throw new CliArgumentsException("Must specify name of new Azure Active Directory application with --aad-name parameter.",
+                    new CliArgument { Name = "app-name", Description = "Name of new Azure Active Directory application" });
+            }
+
+            if (CommandChecker.CommandExists("az"))
+            {
+                List<string> replyUrls;
+                string tempFile, clientSecret, query = CreateQuery(AADName, appName, out tempFile, out clientSecret, out replyUrls);
+
+                ColoredConsole.WriteLine("Query successfully constructed. Creating new Azure AD Application now..");
+
+                var az = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+                       ? new Executable("cmd", $"/c az ad app create {query}")
+                       : new Executable("az", $"ad app create {query}");
+                var stdout = new StringBuilder();
+                var stderr = new StringBuilder();
+
+                int exitCode = await az.RunAsync(o => stdout.AppendLine(o), e => stderr.AppendLine(e));               
+
+                // Clean up file we created to pass data in proper format to az CLI
+                File.Delete($"{tempFile}");
+
+                if (exitCode != 0)
+                {
+                    ColoredConsole.WriteLine(Red(stderr.ToString().Trim(' ', '\n', '\r', '"')));
+                    return;
```

Comment:
do you wanna throw here? if you don't throw on an error, the exit code of the cli process won't indicate an error. It's useful when people are scripting the CLI and then try to check exit code for any errors before proceeding.